video2dataset/workers/optical_flow_worker.py

```diff
@@ -0,0 +1,255 @@
+"""
+Worker for optical flow stage
+"""
+import time
+import pyarrow as pa
+import traceback
+import io
+import numpy as np
+import fsspec
+import tempfile
+import os
+import cv2
+
+from video2dataset.logger import CappedCounter, write_stats
+from video2dataset.subsamplers import OpticalFlowSubsampler
+from video2dataset.dataloader import get_video_dataset
+
+
+def numpy_npz_dumps(numpy_dict):
+    """
+    Dump a dictionary of numpy arrays into a bytestring using numpy npz format.
+
+    Args:
+        numpy_dict (dict): A dictionary containing numpy arrays as values.
+
+    Returns:
+        bytes: A bytestring representing the compressed numpy arrays.
+    """
+
+    stream = io.BytesIO()
+    np.savez_compressed(stream, **numpy_dict)
+    return stream.getvalue()
+
+
+def frames_to_mp4_bytes(frames, fps=30, codec="mp4v"):
```

Comment:
why do we need this?

Reply:
i cant leave the array of frames in streams it wont get written to mp4  -- i need to re-encode as bytes before i pass to the sample writer i thought